hoomd/mpcd/update.py

```diff
@@ -0,0 +1,132 @@
+    `Müller-Plathe <https://doi.org/10.1103/PhysRevE.59.4894>`_,
+    only the fastest particle and the slowest particle are swapped. This is
+    achieved by setting `num_swaps` to 1 and `target_momentum` to a
+    sufficiently large value. It is also feasible for particles that are not
+    necessarily the fastest nor slowest to be selected for momentum exchange
+    as `Tenney and Maginn <https://doi.org/10.1063/1.3276454>`_ did. This is
+    accomplished by setting the `target_momentum` argument to a smaller value.
+
+    This swapping of particle momenta imposes a momentum flux on the system,
+    which drives the flow.
+
+    .. rubric:: Example:
+
+    .. code-block:: python
+
+            flow = hoomd.mpcd.update.ReverseNonequilibriumShearFlow(
+                trigger=20, num_swaps=10, slab_width=0.25, target_momentum=5.0)
+            simulation.operations.updaters.append(flow)
+
+    Attributes:
+        trigger (hoomd.trigger.trigger_like): Trigger to swap momentum.
+
+            .. rubric:: Example:
+
+            .. code-block:: python
+
+                flow.trigger = 20
+
+        num_swaps (int): Maximum number of particle pairs to swap.
+
+            .. rubric:: Example:
+
+            .. code-block:: python
+
+                flow.num_swaps = 10
+
+        slab_width (float): Thickness of momentum-exchange slabs.
+
+            .. rubric:: Example:
+
+            .. code-block:: python
+
+                flow.slab_width = 0.25
+
+        target_momentum (float): Target momentum for swapped particles.
+
+            .. rubric:: Example:
+
+            .. code-block:: python
+
+                flow.target_momentum = 5.0
+
+    """
+
+    # Constructor
+    def __init__(self,
+                 trigger,
+                 num_swaps,
+                 slab_width,
+                 target_momentum=math.inf):
+
+        # Call the parent constructor with the trigger parameter
+        super().__init__(trigger)
+
+        # Create a ParameterDict with the given parameters
+        param_dict = ParameterDict(num_swaps=int(num_swaps),
+                                   slab_width=float(slab_width),
+                                   target_momentum=OnlyTypes(
+                                       float, preprocess=positive_real))
+        param_dict["target_momentum"] = target_momentum
+
+        # Update the internal parameter dictionary created by the parent class
+        self._param_dict.update(param_dict)
+
+    # Use the _attach_hook method to create the C++ version of the object
+    def _attach_hook(self):
+        sim = self._simulation
+        self._cpp_obj = _mpcd.ReverseNonequilibriumShearFlow(
+            sim.state._cpp_sys_def, self.trigger, self.num_swaps,
+            self.slab_width, self.target_momentum)
+
+    @log(category="scalar", requires_run=True)
+    def summed_exchanged_momentum(self):
+        r"""Returns the cumulated momentum exchanged between particles."""
```

Comment:
```suggestion
        r"""float: Total momentum exchanged."""
```
This docstring should also be expanded to explain more what this quantity is.